test/io/ferguson/read-class-reorder.chpl

```diff
@@ -0,0 +1,29 @@
+/*class Base {
+}
+
+class Parent : Base {
+  var x: int;
+  var y: int;
+}
+*/
+
+class Child /*: Parent*/ {
```

Comment:
It's not so clear to me how valuable it is as a feature.... I think I'd prefer to wait for someone to say that they care about it. I've added a test for reordering within parent/child.